libminifi/include/core/state/Value.h

```diff
@@ -74,94 +157,215 @@ class BoolValue : public Value {
+    return false;
+  }
+
+  virtual bool getValue(int64_t &ref) {
+    ref = value;
+    return true;
+  }
+
+  virtual bool getValue(uint64_t &ref) {
+    if (value >= 0) {
+      ref = value;
+      return true;
+    }
+    return true;
+  }
+
+  virtual bool getValue(bool &ref) {
+    return false;
+  }
 
-static inline std::shared_ptr<Value> createValue(
-    const bool &object) {
+  int64_t value;
+};
+
+static inline std::shared_ptr<Value> createValue(const bool &object) {
   return std::make_shared<BoolValue>(object);
 }
 
-static inline std::shared_ptr<Value> createValue(
-    const char *object) {
+static inline std::shared_ptr<Value> createValue(const char *object) {
   return std::make_shared<Value>(object);
 }
 
-static inline std::shared_ptr<Value> createValue(
-    char *object) {
+static inline std::shared_ptr<Value> createValue(char *object) {
   return std::make_shared<Value>(std::string(object));
 }
 
-static inline std::shared_ptr<Value> createValue(
-    const std::string &object) {
+static inline std::shared_ptr<Value> createValue(const std::string &object) {
   return std::make_shared<Value>(object);
 }
 
-
-static inline std::shared_ptr<Value> createValue(
-    const uint32_t &object) {
-  return std::make_shared<Int64Value>(object);
+static inline std::shared_ptr<Value> createValue(const uint32_t &object) {
+  return std::make_shared<UInt64Value>(object);
+}
+#if ( defined(__APPLE__) || defined(__MACH__) || defined(DARWIN) )
+static inline std::shared_ptr<Value> createValue(const size_t &object) {
+  return std::make_shared<UInt64Value>(object);
+}
+#endif
+static inline std::shared_ptr<Value> createValue(const uint64_t &object) {
+  return std::make_shared<UInt64Value>(object);
 }
-static inline std::shared_ptr<Value> createValue(
-    const uint64_t &object) {
+
+static inline std::shared_ptr<Value> createValue(const int64_t &object) {
   return std::make_shared<Int64Value>(object);
 }
 
-static inline std::shared_ptr<Value> createValue(
-    const int &object) {
+static inline std::shared_ptr<Value> createValue(const int &object) {
   return std::make_shared<IntValue>(object);
 }
 
-
 /**
  * Purpose: ValueNode is the AST container for a value
  */
 class ValueNode {
  public:
   ValueNode()
       : value_(nullptr) {
-
   }
 
+  ValueNode(ValueNode &&vn) = default;
+  ValueNode(const ValueNode &vn) = default;
+
   /**
    * Define the representations and eventual storage relationships through
    * createValue
    */
   template<typename T>
-  auto operator=(
-      const T ref) -> typename std::enable_if<std::is_same<T, int >::value ||
-      std::is_same<T, uint32_t >::value ||
-      std::is_same<T, uint64_t >::value ||
-      std::is_same<T, bool >::value ||
-      std::is_same<T, char* >::value ||
-      std::is_same<T, const char* >::value ||
-      std::is_same<T, std::string>::value,ValueNode&>::type {
+  auto operator=(const T ref) -> typename std::enable_if<std::is_same<T, int >::value ||
```

Comment:
Guess std::is_integral<T> would help here reducing this.

Reply:
There were a few excluded types. Once I get around to implementing those I'd prefer to use integral too